pkg/operator/ceph/file/mds/mds.go

```diff
@@ -127,10 +128,20 @@ func (c *Cluster) Start() error {
 	// Always create double the number of metadata servers to have standby mdses available
 	replicas := c.fs.Spec.MetadataServer.ActiveCount * 2
 
+	mdsToSkipReconcile, err := controller.SkipReconcile(c.clusterInfo.Context, c.context, c.clusterInfo.Namespace, config.MdsType, AppName)
+	if err != nil {
+		return errors.Wrap(err, "failed to check for mds to skip reconcile")
+	}
+
 	// keep list of deployments we want so unwanted ones can be deleted later
 	desiredDeployments := map[string]bool{} // improvised set
 	// Create/update deployments
 	for i := 0; i < int(replicas); i++ {
+		if mdsToSkipReconcile.Has(k8sutil.IndexToName(i)) {
```

Comment:
`k8sutil.IndexToName(i)` method will be called thrice inside the for loop. How about we saving it in a variable and then using that variable.

Reply:
We need to call `k8sutil.IndexToName(i)` everytime for each mds deployment. 

https://github.com/rook/rook/blob/55f4f7d69bb6415ebba6f5ae8565b14d8e28d92c/pkg/operator/ceph/controller/spec.go#L948-L953
 
above code is from my branch, if you see we only append in the set if it matches the `daemonID`, so we need to call `k8sutil.Index()` everytime.

Reply:
sorry still don't get why we need to call it everytime. 
If we invoke it once after line 139
`daemonName := k8sutil.IndexToName(i)`
will that not work?

Reply:
I read your comment completely wrong, I thought the comment was about why using `k8sutil.IndexToName(i)` in the loop. But you mentioned it clearly and I read the comment again it was clear. I have made the changes fresh mind today helped  thanks :XD